tests/lang/core.pk

```diff
@@ -30,6 +30,12 @@ def fn(p1, p2, p3) end
 assert(fn.name == "fn")
 assert(fn.arity == 3)
 
+## String functions
+assert(str_sub('c programming', 2, 11) == 'programming')
+assert(str_sub('foobarbaz', 2, 3) == 'oba')
+assert(str_sub('abcdef', 1, 2) == 'bc')
+assert(str_sub('I am a boy', 0, 4) == 'I am')
+
```

Comment:
Looks good, but add ...
- empty string to the tests as well to cover all the edge cases.
- Also zero length substrings (`== ''`) 
- try the position equals to the string length and len = 0